scripts/main.js

```diff
@@ -0,0 +1,79 @@
+const bookCard = document.getElementById('book-card')
+
+const myLibrary = []
+
+function Book(author, title, pages, read = false) {
+    this.id = Math.floor(Math.random() * 100)
+    this.author = author
+    this.title = title
+    this.pages = pages
+    this.read = read
+}
+
+function addBookToLibrary(author, title, pages, read) {
+    myLibrary.push(new Book(author, title, pages, read))
+    console.log(myLibrary)
+}
+
+function render() {
+    bookCard.innerHTML = ''
+    for (const book of myLibrary) {
+        const bookItem = `<span>Author: ${book.author}</span> <span>Title: ${book.title}</span> <span>Pages: ${book.pages} </span> <span>Read: ${book.read}</span>`
+        const bookElem = document.createElement('div')
+        bookElem.setAttribute('id', `${book.id}`)
+        const button = document.createElement('button')
+        const toggleBtn = document.createElement('button')
+        button.setAttribute('onclick', 'removeBook(this)')
+        toggleBtn.setAttribute('onclick', 'toggleReadStatus(this)')
+        book.read ? toggleBtn.classList.add('toggle-class') : toggleBtn;
+
+        button.textContent = ' x'
+        toggleBtn.textContent = 'Change Read Status'
+        bookElem.innerHTML = bookItem
+        bookElem.appendChild(button)
+        bookElem.appendChild(toggleBtn)
+        bookCard.appendChild(bookElem)
+    }
+}
+
+const saveBtn = document.getElementById('save')
+
+function addBook(e) {
+    const author = document.getElementById('author').value
+    const title = document.getElementById('title').value
+    const pages = document.getElementById('pages').value
+    const read = document.getElementById('read').value === 'yes'
+    e.preventDefault()
+    addBookToLibrary(author, title, pages, read)
+    document.getElementById('save').style.display = 'none'
+    render()
+}
+
+function removeBook(id) {
+    const test = myLibrary.find((element) => element.id == id.parentElement.id)
+    myLibrary.splice(myLibrary.indexOf(test), 1)
+    render()
+}
+
+function toggleReadStatus(id) {
+    const test = myLibrary.find((element) => element.id === id.parentElement.id)
+    const readStatus = myLibrary[myLibrary.indexOf(test)].read
+    myLibrary[myLibrary.indexOf(test)].read = !myLibrary[myLibrary.indexOf(test)].read
+    render()
+}
```

Comment:
Could you investigate this method? There seems to be a bug 🐛 here affecting the ability of the app to toggle read status.

- Why not used [`Array.findIndex()`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) method here? 🤔 
Since it will return the `index` of the array if available. This way, you can always check if the index is four before changing the read status.

- `const readStatus = myLibrary[myLibrary.indexOf(test)].read` also seems to be redundant (unused).

See demo below:

![record1](https://user-images.githubusercontent.com/36057474/82664722-3af64900-9c2a-11ea-900d-f1b318556050.gif)